autogen/agentchat/contrib/capabilities/vision_capability.py

```diff
@@ -0,0 +1,172 @@
+from typing import Dict, List, Optional, Union
+
+from autogen.agentchat.assistant_agent import ConversableAgent
+from autogen.agentchat.contrib.capabilities.agent_capability import AgentCapability
+from autogen.agentchat.contrib.img_utils import (
+    convert_base64_to_data_uri,
+    get_image_data,
+    get_pil_image,
+    gpt4v_formatter,
+    message_formatter_pil_to_b64,
+)
+from autogen.agentchat.contrib.multimodal_conversable_agent import MultimodalConversableAgent
+from autogen.agentchat.conversable_agent import colored
+from autogen.code_utils import content_str
+from autogen.oai.client import OpenAIWrapper
+
+
+class VisionCapability(AgentCapability):
+    """
+        We can add vision capability to regular ConversableAgent, even if the agent does not have the multimodal capability,
+    such as GPT-3.5-turbo agent, Llama, Orca, or Mistral agents. This vision capability will invoke a LMM client to describe
+    the image (captioning) before sending the information to the agent's actual client.
+
+        The vision capability will hook to the ConversableAgent's `process_last_received_message`.
+
+        Some technical details:
+        When the agent (who has the vision capability) received an message, it will:
+        1. _process_received_message:
+            a. _append_oai_message
+        2. generate_reply: if the agent is a MultimodalAgent, it will also use the image tag.
+            a. hook process_last_received_message (NOTE: this is where the vision capability will be hooked to.)
+            b. hook process_all_messages_before_reply
+        3. send:
+            a. hook process_message_before_send
+            b. _append_oai_message
+    """
+
+    def __init__(
+        self,
+        lmm_config: Dict,
+        description_prompt: Optional[str] = "Describe the following image in details.",
+    ):
+        """
+        Args:
+            lmm_config (dict or False): LMM (multimodal) client configuration,
+                which will be used to call LMM to describe the image.
+            description_prompt (str, optional): The prompt to use for describing the image.
+        """
+        assert lmm_config, "Vision Capability requires a valid lmm_config."
+        self._lmm_config = lmm_config
+        self._description_prompt = description_prompt
+        self._lmm_client = OpenAIWrapper(**lmm_config)
+
+    def add_to_agent(self, agent: ConversableAgent):
```

Comment:
Would it also be useful for `add_to_agent` to append extra info to the base agent's system message? Something like:

"You've been given the ability to interpret images."

Maybe that would make the agent less likely to deny having that ability.